applications/rpc/rpc_gui.c

```diff
@@ -0,0 +1,137 @@
+#include "flipper.pb.h"
+#include "rpc_i.h"
+#include "gui.pb.h"
+#include <gui/gui_i.h>
+
+void rpc_system_gui_screen_frame_callback(uint8_t* data, size_t size, void* context) {
+    furi_assert(data);
+    furi_assert(size == 1024);
+    furi_assert(context);
+
+    Rpc* rpc = context;
+
+    PB_Main* frame = furi_alloc(sizeof(PB_Main));
+
+    frame->which_content = PB_Main_gui_screen_stream_frame_tag;
+    frame->command_status = PB_CommandStatus_OK;
+    frame->content.gui_screen_stream_frame.data = furi_alloc(PB_BYTES_ARRAY_T_ALLOCSIZE(size));
+    uint8_t* buffer = frame->content.gui_screen_stream_frame.data->bytes;
+    uint16_t* frame_size_msg = &frame->content.gui_screen_stream_frame.data->size;
+    *frame_size_msg = size;
+    memcpy(buffer, data, size);
+
+    rpc_send_and_release(rpc, frame);
+
+    free(frame);
+}
+
+void rpc_system_gui_start_screen_stream_process(const PB_Main* request, void* context) {
+    rpc_send_and_release_empty(context, request->command_id, PB_CommandStatus_OK);
+
+    Gui* gui = furi_record_open("gui");
+    gui_set_framebuffer_callback(gui, rpc_system_gui_screen_frame_callback, context);
+    furi_record_close("gui");
+}
+
+void rpc_system_gui_stop_screen_stream_process(const PB_Main* request, void* context) {
+    rpc_send_and_release_empty(context, request->command_id, PB_CommandStatus_OK);
+
+    Gui* gui = furi_record_open("gui");
+    gui_set_framebuffer_callback(gui, NULL, NULL);
+    furi_record_close("gui");
```

Comment:
Closing record means that record provider can exit safely. Don't close record till you using it(directly or indirectly)